File: AndroidTestingBlueprint/app/src/test/java/com/example/android/testing/blueprint/LocalUnitTest.java
/*
 * Copyright 2015, The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.android.testing.blueprint;

import android.app.Activity;
import android.app.Application;
import android.bluetooth.BluetoothAdapter;

import org.junit.Test;

import java.io.File;

import static junit.framework.TestCase.assertTrue;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.notNullValue;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;
import static org.mockito.Mockito.when;

/**
 * A test class with unit tests that mock classes from the Android framework.
 */
public class LocalUnitTest {

    @Test
    public void mockFinalMethod() {
        Activity activity = mock(Activity.class);
        Application app = mock(Application.class);
        when(activity.getApplication()).thenReturn(app);

        assertThat(app, is(equalTo(activity.getApplication())));

        verify(activity).getApplication();
        verifyNoMoreInteractions(activity);
    }

    @Test
    public void mockFinalClass() {
        BluetoothAdapter adapter = mock(BluetoothAdapter.class);
        when(adapter.isEnabled()).thenReturn(true);

        assertTrue(adapter.isEnabled());

        verify(adapter).isEnabled();
        verifyNoMoreInteractions(adapter);
    }

    @Test
    public void loadTestDataFromResources() {
        File helloBleprintJson = new File(getClass().getResource("/helloBlueprint.json").getPath());
        assertThat(helloBleprintJson, notNullValue());
    }

}

File: AndroidTestingBlueprint-kotlinApp/app/src/test/java/com/example/android/testing/blueprint/LocalUnitTest.kt
/*
 * Copyright 2017, The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.android.testing.blueprint

import android.app.Activity
import android.app.Application
import android.bluetooth.BluetoothAdapter
import junit.framework.TestCase.assertTrue
import org.hamcrest.CoreMatchers.notNullValue
import org.junit.Assert.assertEquals
import org.junit.Assert.assertThat
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify
import org.mockito.Mockito.verifyNoMoreInteractions
import java.io.File
import org.hamcrest.CoreMatchers.`is` as matchIs
import org.mockito.Mockito.`when` as mockWhen

/**
 * A test class with unit tests that mock classes from the Android framework.
 */
class LocalUnitTest {

    @Test fun mockFinalMethod() {
        val activity = mock(Activity::class.java)
        val app = mock(Application::class.java)
        mockWhen(activity.application).thenReturn(app)

        assertEquals(app, activity.application)

        verify(activity).application
        verifyNoMoreInteractions(activity)
    }

    @Test fun mockFinalClass() {
        val adapter = mock(BluetoothAdapter::class.java)
        mockWhen(adapter.isEnabled).thenReturn(true)

        assertTrue(adapter.isEnabled)

        verify(adapter).isEnabled
        verifyNoMoreInteractions(adapter)
    }

    @Test fun loadTestDataFromResources() {
        val helloBlueprintJson = File(javaClass.getResource("/helloBlueprint.json").path)
        assertThat(helloBlueprintJson, notNullValue())
    }

}
